skbio/stats/composition.py

```diff
@@ -0,0 +1,419 @@
+constant and there are :math:`D` components for each composition. In this
+module :math:`c=1`. Compositional data can be analyzed using Aitchison
+geometry [1]_
+
+However, in this framework, standard real Euclidean operations such as addition
+and multiplication no longer apply.  Only operations such as perturbation and
+power can be used to manipulate this data [1]_
+
+This module allows two styles of manipulation of compositional data.
+Compositional data can be analyzed using perturbation and power operations,
+which can be useful for simulation studies. The alternative strategy is to
+transform compositional data into the real space.  Right now, the centre log
+ratio transform (clr) [1]_ can be used to accomplish this.  This transform can
+be useful for performing standard statistical tools such as parametric
+hypothesis testing, regressions and more.
+
+The major caveat of using this framework is dealing with zeros.
+In the Aitchison geometry, only compositions with nonzero components can be
+considered. The multiplicative replacement technique [2]_ can be used to
+substitute these zeros with small pseudocounts without introducing major
+distortions to the data.
+
+Functions
+---------
+
+.. autosummary::
+   :toctree: generated/
+
+   multiplicative_replacement
+   perturb
+   perturb_inv
+   power
+   clr
+   centralize
+
+Examples
+--------
+
+>>> import numpy as np
+
+Consider a very simple environment with only 3 species.
+The species in the environment are equally distributed and their
+proportions are equilvalent:
+
+>>> otus = np.array([1./3, 1./3., 1./3])
+
+Suppose that an antibiotic kills off half of the population for the
+first two species, but doesn't harm the third species.
+Then the perturbation vector would be as follows
+
+>>> antibiotic = np.array([1./2, 1./2, 1])
+
+And the resulting perturbation would be
+
+>>> perturb(otus, antibiotic)
+array([ 0.25,  0.25,  0.5 ])
+
+Reference
+---------
+.. [1] V. Pawlowsky-Glahn. "Lecture Notes on Compositional Data Analysis"
+.. [2] J. A. Martin-Fernandez. "Dealing With Zeros and Missing Values in
+       Compositional Data Sets Using Nonparametric Imputation"
+
+"""
+from __future__ import absolute_import, division, print_function
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+import scipy.stats as ss
+
+
+def _closure(mat):
+    """
+    Performs closure to ensure that all elements add up to 1
+    mat : array-like
+       a matrix of proportions where
+       rows = compositions
+       columns = components
+
+    Returns
+    -------
+    mat_ : numpy.ndarray, np.float64
```

Comment:
Actually, I don't know how that shows in the html docs. It is probably fine, though I personally don't find the dtype that important, particularly because it's the most common one.